CONTRIBUTING.md

```diff
@@ -122,6 +122,13 @@ which is important if it shows an example running against live resources that li
 
 More guidance for writing documentation tests can be found in our [Rust Guidelines].
 
+### Performance testing
```

Comment:
1. Use same casing as other titles.
2. This is less important to most devs than debugging. Keep as an H3 header but at the bottom of this `## Testing` section i.e., right above the `## Code Review Process` heading.